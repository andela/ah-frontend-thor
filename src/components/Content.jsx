import React from "react";

const Content = () => {
	return (
		<div className="container">
			<form>
				<div className="form-group">
					<input
						type="email"
						className="form-control"
						placeholder="Enter email"
					/>
				</div>
				<div className="form-group">
					<input
						type="password"
						className="form-control"
						placeholder="Enter Password"
					/>
				</div>
				<button type="submit" className="btn btn-primary">
					Login
				</button>
			</form>
		</div>
	);
};
export default Content;
