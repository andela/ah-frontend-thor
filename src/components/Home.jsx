import React from "react";

const Home = () => (
  <div>
    Author's Haven
  </div>
);
export default Home;
