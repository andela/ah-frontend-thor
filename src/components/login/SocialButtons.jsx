import React from "react";

import Google from "./Google";
import Facebook from "./Facebook";
import { GOOGLE_URL, FB_URL } from "../../../config";

const Home = () => (
  <div>
    <Facebook url={FB_URL} />
    <Google url={GOOGLE_URL} />
  </div>
);

export default Home;
