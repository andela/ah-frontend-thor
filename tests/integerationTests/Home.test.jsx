import "babel-polyfill";
import React from "react";
import { mount, shallow } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

// eslint-disable-next-line import/no-unresolved
import Content from "../../src/components/Contents";
// eslint-disable-next-line import/no-unresolved
import SocialButtons from "../../src/components/login/SocialButtons";
import Home from "../../src/components/Home";

describe("<Content />", () => {
  document
    .getElementsByTagName("head")[0]
    .appendChild(document.createElement("script"));
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  const initialUserState = {
    social: {
      isLoggedIn: false,
      user: null,
      token: null
    }
  };

  const store = mockStore({ ...initialUserState });
  const provider = shallow(
    <Provider store={store}>
      <Content />
    </Provider>
  );
  const home = shallow(<Content />);
  const social = shallow(<SocialButtons />);

  it("renders <Provider/> correctly", () => {
    expect(provider).toMatchSnapshot();
  });
  it("renders <Provider/> correctly", () => {
    expect(social).toMatchSnapshot();
  });

  it("renders <Home/> correctly", () => {
    expect(home).toMatchSnapshot();
  });
  describe("<Content />", () => {
    it("should render without crashing", () => {
      expect(mount.bind(null, <Provider store={store}><Content /></Provider>)).not.toThrow();
    });
  });
  
  it("should render container class", () => {
    const editor = shallow(<Content />);
    expect(editor.find("div.container").length).toEqual(1);
  });
  it("should render <Facebook/>", () => {
    expect(shallow(<SocialButtons />).find("Connect(Facebook)")).toHaveLength(1);
  });
  it("should render <Google/>", () => {
    expect(shallow(<SocialButtons />).find("Connect(Google)")).toHaveLength(1);
  });
  it("tests the footer of the app against the snapshot", () => {
    const component = mount(<Home />);
    expect(component).toMatchSnapshot();
  });
});