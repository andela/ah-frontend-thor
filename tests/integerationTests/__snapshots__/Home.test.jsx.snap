// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the mounted Footer component tests the footer of the app against the snapshot 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": Object {
      "instance": <div>
        <h3>
          Author's Haven
        </h3>
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h3>
          Author's Haven
        </h3>,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <h3>
            Author's Haven
          </h3>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Author's Haven",
          },
          "ref": null,
          "rendered": Array [
            "Author's Haven",
          ],
          "type": "h3",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": Object {
        "instance": <div>
          <h3>
            Author's Haven
          </h3>
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h3>
            Author's Haven
          </h3>,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <h3>
              Author's Haven
            </h3>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Author's Haven",
            },
            "ref": null,
            "rendered": Array [
              "Author's Haven",
            ],
            "type": "h3",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
